test/csuite/wt12945_flags_eviction_prefetch/main.c

```diff
@@ -0,0 +1,217 @@
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "test_util.h"
+
+/*
+ * JIRA ticket reference: WT-12945 Test case description: This is a test case that looks for crashes
+ * when prefetch and eviction of the same page happens at the same time.
+ *
+ * This variant tests dirty eviction.
+ */
+
+/* TEMPORARY: NOTE: This is loosely based on wt-2535_insert_race. */
+
+/* Constants */
+
+#define NUM_WARM_UP_RECORDS 100000 /* How many records to insert during warmup. */
+/* First record to change to give pre-fetch thread time to begin prefetching */
+#define FIRST_RECORD_TO_CHANGE 2000
+#define NUM_EVICTION 1 /* How many times to force eviction */
+
+void *thread_do_prefetch(void *);
+
+static uint64_t ready_counter;
+
+/*
+ * set_key --
+ *     Wrapper providing the correct typing for the WT_CURSOR::set_key variadic argument.
+ */
+static void
+set_key(WT_CURSOR *cursor, uint64_t value)
+{
+    cursor->set_key(cursor, value);
+}
+
+/*
+ * set_value --
+ *     Wrapper providing the correct typing for the WT_CURSOR::set_value variadic argument.
+ */
+static void
+set_value(TEST_OPTS *opts, WT_CURSOR *cursor, uint64_t value)
+{
+    WT_UNUSED(opts);
+    cursor->set_value(cursor, value);
+}
+
+/*
+ * get_value --
+ *     Wrapper providing the correct typing for the WT_CURSOR::get_value variadic argument.
+ */
+static uint64_t
+get_value(TEST_OPTS *opts, WT_CURSOR *cursor)
+{
+    uint64_t value64;
+    WT_UNUSED(opts);
+
+    testutil_check(cursor->get_value(cursor, &value64));
+    return (value64);
+}
+
+/*
+ * main --
+ *     TODO: Add a comment describing this function.
+ */
+int
+main(int argc, char *argv[])
+{
+    TEST_OPTS *opts, _opts;
+    WT_CURSOR *cursor;
+    WT_SESSION *wt_session;
+    pthread_t prefetch_thread_id;
+    uint64_t ready_counter_local;
+    uint64_t record_idx;
+
+    opts = &_opts;
+    memset(opts, 0, sizeof(*opts));
+    opts->nthreads = 1;                   /* Value ignored */
+    opts->nrecords = NUM_WARM_UP_RECORDS; /* Use: How many records to insert during warmup. */
+    opts->table_type = TABLE_ROW;         /* Value ignored */
+    testutil_check(testutil_parse_opts(argc, argv, opts));
+    opts->nthreads = 1;           /* Force desired value */
+    opts->table_type = TABLE_ROW; /* Force desired value */
+    testutil_recreate_dir(opts->home);
+
+    testutil_check(wiredtiger_open(opts->home, NULL,
+      "create,cache_size=2G,eviction=(threads_max=5),statistics=(all),statistics_log=(json,on_"
+      "close,wait=1)",
+      &opts->conn));
+
+    /* Create the session for eviction. */
+    testutil_check(opts->conn->open_session(opts->conn, NULL, NULL, &wt_session));
+    testutil_check(
+      wt_session->create(wt_session, opts->uri, "key_format=Q,value_format=Q,leaf_page_max=32k"));
+
+    /* Warm-up: Insert documents until earlier documents are forced out of the cache */
```

Comment:
This look similar to how I expected, but this section is non deterministic as it assumes we'll do enough insertions to trigger eviction of a significant part of the tree. We could always ensure all this content is evicted by looping through the tree with an eviction cursor and evicting every Nth key.